package/MDAnalysis/analysis/hydrogenbonds/hbond_autocorrel.py

```diff
@@ -321,11 +321,15 @@ def __init__(self, universe,
             raise ValueError("Donors and Hydrogen groups must be identical "
                              "length.  Try using `find_hydrogen_donors`.")
 
-        self.exclusions = exclusions
-        if self.exclusions:
-            if not len(self.exclusions[0]) == len(self.exclusions[1]):
+        if exclusions is not None:
+            if len(exclusions[0]) != len(exclusions[1]):
                 raise ValueError(
-                    "'exclusion' must be two arrays of identical length")
+                        "'exclusion' must be two arrays of identical length")
+            self.exclusions = np.column_stack((exclusions[0], 
+                                               exclusions[1]))
+                                               .astype(np.intp)
```

Comment:
This syntax is invalid and causes the tests to break. Here, `.astype(np.intp)` is taken as a new instruction rather than the continuation of the previous line.

There are various ways to cut this line not to be too long. None is ideal but I would suggest something like:

```python
            self.exclusions = np.column_stack((
                exclusions[0], exclusions[1],
            )).astype(np.intp)
````

Note that I can cut the line where I did because python omits the end of lines until it closes the parentheses. You cannot cut where you did because the line is caught up on parentheses before the dot, and python does not loog ahead to see the dot.